Quickref.md

```diff
@@ -0,0 +1,168 @@
+
+-- Event to Event via monadic function
+[ ]   push       :: (a -> m (Maybe b)) -> Event a -> Event b
+[ ]   pushAlways :: (a -> m        b ) -> Event a -> Event b
+      -- Note supplied function operates in [H] context
+
+-- Event to monadic Event
+[H]   headE     :: Event a -> m (Event a)
+[H]   tailE     :: Event a -> m (Event a)
+[H]   headTailE :: Event a -> m (Event a, Event a)
+```
+
+## Functions producing Behavior
+
+```haskell
+-- Trivial Behavior
+[ ]   constant ::              a ->    Behavior a
+
+-- Extract Behavior from Dynamic
+[ ]   current  ::      Dynamic a ->    Behavior a
+
+-- Event to monadic Behavior
+[H]   hold     :: a ->   Event a -> m (Behavior a)
+
+-- Transform Behavior to Behavior using function
+[ ]   fmap ::   (a -> b) -> Behavior a -> Behavior b
+[ ]   ffor :: Behavior a ->   (a -> b) -> Behavior b
+[ ]   <*> :: Behavior (a -> b) -> Behavior a -> Behavior b
+[ ]   >>= :: Behavior a -> (a -> Behavior b) -> Behavior b
+[ ]   <> :: Monoid a => Behavior a -> Behavior a -> Behavior a
+-- ... plus many more due to typeclass membership
+
+-- Behavior to Behavior via monadic value
+[S]   sample :: Behavior a -> m a
+[ ]   pull   ::               m a -> Behavior a
+      -- Note supplied value is in [S] context
+```
+
+## Functions producing Dynamic
+
+```haskell
+-- Trivial Dynamic
+[ ]   constDyn ::                                  a            ->    Dynamic a
+
+-- Construct Dynamic from Event
+[H]   holdDyn       ::                             a -> Event a -> m (Dynamic a)
+[H]   foldDyn       :: (a -> b ->           b ) -> b -> Event a -> m (Dynamic b)
+[H]   foldDynMaybe  :: (a -> b ->     Maybe b ) -> b -> Event a -> m (Dynamic b)
+[H]   foldDynM      :: (a -> b -> m'        b ) -> b -> Event a -> m (Dynamic b)
+[H]   foldDynMaybeM :: (a -> b -> m' (Maybe b)) -> b -> Event a -> m (Dynamic b)
+      -- Note m' supplies [H] context
+[H]   count :: Num b => Event a -> m (Dynamic b)
+[H]   toggle :: Bool -> Event a -> m (Dynamic Bool)
+
+-- Transform Dynamic to Dynamic using function
+[H]   forDyn  ::   Dynamic a ->  (a -> b) -> m (Dynamic b)
+[H]   mapDyn  :: (a ->    b) -> Dynamic a -> m (Dynamic b)
+[H]   mapDynM :: (a -> m' b) -> Dynamic a -> m (Dynamic b)
+      -- Note m' supplies [S] context
+[H]   splitDyn :: Dynamic (a, b) -> m (Dynamic a, Dynamic b)
+
+-- Combine multiple Dynamics
+[H]   mconcatDyn :: Monoid a => [Dynamic a] -> m (Dynamic a)
+[H]   distributeDMapOverDyn :: GCompare k => DMap (WrapArg Dynamic k) -> m (Dynamic (DMap k))
+[H]   combineDyn :: (a -> b -> c) -> Dynamic a -> Dynamic b -> m (Dynamic c)
+
+-- Efficient one-to-many fanout
+[ ]   demux      :: Ord k => Dynamic k -> Demux k
+[H]   getDemuxed ::  Eq k =>              Demux k -> k -> m (Dynamic Bool)
+
+-- Dynamic to Dynamic, removing updates w/o value change
+[ ]   nubDyn :: Eq a => Dynamic a -> Dynamic a
+
+-- Dynamic to identical Dynamic with debug trace.
```

Comment:
Note that with any of the trace functions, the value will only be traced if the output is actually used; for traceDyn and traceEvent, that means it only traces when someone's actually listening to the output event.  For traceBehavior, it means that it'll only trace when the Behavior's value is actually sampled.